lib/data/network_manager.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:connectivity/connectivity.dart';
+
+///abstract class to mock it during tests
+abstract class NetworkManagerAPI {
```

Comment:
what does -API postfix mean here?
this one is responsible for connectivity checking, right?
let's name it smth like ConnectivityManager or ConnectivityChecker and rename it's implementation correspondingly.

Reply:
It's just another way how to name interface to describe what we have to implement. I don't like INetworkManager cause of it's even more ugly than NetworkManagerAPI. 
'Rename it's implementation correspondingly' - what exactly to you mean? 
MobileConnectivityManager and MockConnectivityManager?

Reply:
I just don't see any reason to add API postfix here